src/vs/workbench/contrib/debug/browser/loadedScriptsPicker.ts

```diff
@@ -0,0 +1,114 @@
+	const editorService = accessor.get(IEditorService);
+	const sessions = debugService.getModel().getSessions(false);
+	const modelService = accessor.get(IModelService);
+	const languageService = accessor.get(ILanguageService);
+
+	const localDisposableStore = new DisposableStore();
+	const quickPick = quickInputService.createQuickPick<IPickerLoadedScriptItem>();
+	quickPick.matchOnLabel = quickPick.matchOnDescription = quickPick.matchOnDetail = quickPick.sortByLabel = false;
+	quickPick.placeholder = nls.localize('moveFocusedView.selectView', "Search loaded scripts by name");
+	quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+
+	localDisposableStore.add(quickPick.onDidChangeValue(async () => {
+		quickPick.items = await _getPicks(quickPick.value, sessions, editorService, modelService, languageService);
+	}));
+	localDisposableStore.add(quickPick.onDidAccept(() => {
+		const selectedItem = quickPick.selectedItems[0];
+		selectedItem.accept();
+		localDisposableStore.clear();
+		quickPick.hide();
+		quickPick.dispose();
+	}));
+	quickPick.show();
+}
+
+function _getLabelName(source: Source) {
+	const posOfForwardSlash = source.name.lastIndexOf('/');
+	const posOfBackSlash = source.name.lastIndexOf('\\');
+
+	const delimeterPos = posOfForwardSlash > posOfBackSlash ? posOfForwardSlash : posOfBackSlash;
+	const label = source.name.substring(delimeterPos + 1);
+	return label;
+}
+
+async function _getPicksFromSession(session: IDebugSession, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const items: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+	items.push({ type: 'separator', label: session.name });
+	const sources = await session.getLoadedSources();
+
+	sources.forEach((element: Source) => {
+		const pick = _createPick(element, filter, editorService, modelService, languageService);
+		if (pick) {
+			items.push(pick);
+		}
+
+	});
+	return items;
+}
+async function _getPicks(filter: string, sessions: IDebugSession[], editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): Promise<Array<IPickerLoadedScriptItem | IQuickPickSeparator>> {
+	const loadedScriptPicks: Array<IPickerLoadedScriptItem | IQuickPickSeparator> = [];
+
+
+	const picks = await Promise.all(
+		sessions.map((session) => _getPicksFromSession(session, filter, editorService, modelService, languageService))
+	);
+
+	for (const row of picks) {
+		for (const elem of row) {
+			loadedScriptPicks.push(elem);
+		}
+	}
+	return loadedScriptPicks;
+}
+
+function _createPick(source: Source, filter: string, editorService: IEditorService, modelService: IModelService, languageService: ILanguageService): IPickerLoadedScriptItem | undefined {
+
+	const label = _getLabelName(source);
+	const desc = dirname(source.uri).path;
```

Comment:
It doesn't seem right to use the source name for one part and the source uri for another part. We have the `ILabelService` which knows the right way to format URIs, that is actually what we should use here.

Try using `getUriBasenameLabel` from that and `getUriLabel(dirname(source.uri))` for the description